core/embed/rust/src/ui/display.rs

```diff
@@ -49,11 +49,18 @@ pub fn fade_backlight(target: i32) {
     }
 }
 
-pub fn rect(r: Rect, fg_color: Color) {
+pub fn rect_fill(r: Rect, fg_color: Color) {
     display::bar(r.x0, r.y0, r.width(), r.height(), fg_color.into());
 }
 
-pub fn rounded_rect(r: Rect, fg_color: Color, bg_color: Color, radius: u8) {
+pub fn rect_stroke(r: Rect, fg_color: Color) {
+    display::bar(r.x0, r.y0, r.width(), 1, fg_color.into());
+    display::bar(r.x0, r.y0 + r.height() - 1, r.width(), 1, fg_color.into());
+    display::bar(r.x0, r.y0, 1, r.height(), fg_color.into());
+    display::bar(r.x0 + r.width() - 1, r.y0, 1, r.height(), fg_color.into());
+}
+
+pub fn rect_fill_rounded(r: Rect, fg_color: Color, bg_color: Color, radius: u8) {
```

Comment:
FWIW we should check somewhere that `radius` is 2, 4, 8, or 16. That's what the C code does, but the check is silent. An `assert!` in `display::bar_radius` probably?

Reply:
Added in be6ecd2b9848979c589a79c28b43d0ced9ff376a.